links/link_veth.go

```diff
@@ -106,8 +108,14 @@ func (l *LinkVEth) Deploy(ctx context.Context) error {
 	l.m.Lock()
 	defer l.m.Unlock()
 
-	for _, ep := range l.Endpoints {
-		if ep.GetState() < EndpointDeployStateReady {
+	// since each node calls deploy on its links, we need to make sure that we only deploy
+	// the link once, even if multiple nodes call deploy on the same link.
+	if l.deploymentState == LinkDeploymentStateDeployed {
+		return nil
+	}
+
+	for _, ep := range l.GetEndpoints() {
+		if ep.GetNode().GetState() != "deployed" {
 			return nil
 		}
 	}
```

Comment:
@steiler this seems to match what we have discussed.

1. The link is the entity that gets deployed.
2. The link has a deployment state. If it is already deployed the link deployment function returns. This is when several nodes concurrently attempt to deploy the same link
3. The link is deployed when both nodes it needs are deployed. This is done via `GetState()` method of a Node interface. The state is encoded as string, because we have to find a way where we can store an aliased type without causing a circular dep.

Since several nodes may be assi

Reply:
Can we store that enum under `types`?!
I would want to avoid using plane strings from the beginning.

Reply:
I tried this at first, but types import links (because of link.LinkDefinition).
Some structs replacement would be needed

Reply:
did in [25d99e0](https://github.com/srl-labs/containerlab/pull/1507/commits/25d99e0795ab87507baf4d7a68fb6cf8166f88f7)